opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -726,9 +751,29 @@ namespace detail {
         asImpl().assembleMassBalanceEq(state);
 
         // -------- Well equations ----------
+
         V aliveWells;
-        asImpl().addWellEq(state, well_state, aliveWells);
-        addWellControlEq(state, well_state, aliveWells);
+        const int np = wells().number_of_phases;
```

Comment:
This code may not work if no wells are present, use `wellsActive()` to check first.